### ISSUE TITLE
cnf/hints/linux breaks ccflags

### ISSUE THREAD

**Issue by chneukirchen** [arsv/perl-cross#25]
In cnf/hints/linux, cflags are added with plain +=, which connects them with the previous values:

`ccflags+='-D_POSIX_C_SOURCE -D_XOPEN_SOURCE -D_GNU_SOURCE'`

Care should be taken to add a leading space there.

**Comment by chneukirchen** [comment]
Likewise, in cnf/configure_prog.sh, there is

`appendvar 'ccflags' " -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64"`

and then I see in config.log 

`gcc -D_POSIX_C_SOURCE -D_XOPEN_SOURCE -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64-W -c -o try.o try.c`

(no space before -W)

**Comment by arsv** [comment]
Could you please show your configure arguments?
I just tried `./configure -Duselargefile` and I get a properly separated -W

appendvar adds a space when appending to a non-empty value.
https://github.com/arsv/perl-cross/blob/master/cnf/configure__f.sh#L317-L324
That space before -D_LARGEFILE_SOURCE shouldn't be there actually.
And += in hint files gets translated into appendvar, so same applies there.

**Comment by chneukirchen** [comment]
I used 1.0.3 release on top of 5.24.0. It only happens when cross-compiling.

./configure --prefix=/usr -Dusethreads -Duseshrplib -Dusesoname -Dusevendorprefix -Dprefix=/usr -Dvendorprefix=/usr -Dprivlib=/usr/share/perl5/core_perl -Darchlib=/usr/lib/perl5/core_perl -Dsitelib=/usr/share/perl5/site_perl -Dsitearch=/usr/lib/perl5/site_perl -Dvendorlib=/usr/share/perl5/vendor_perl -Dvendorarch=/usr/lib/perl5/vendor_perl -Dscriptdir=/usr/bin -Dvendorscript=/usr/bin -Dinc_version_list=none -Dman1ext=1p -Dman3ext=3p -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 --target=armv7l-linux-gnueabihf -Dcccdlflags=-fPIC '-Doptimize= -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2  -O2 -pipe -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard   -I/usr/armv7l-linux-gnueabihf/usr/include ' '-Dccflags= -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2  -O2 -pipe -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard   -I/usr/armv7l-linux-gnueabihf/usr/include ' '-Dlddlflags=-shared -specs=/void-packages/common/environment/configure/gccspecs/hardened-ld -Wl,-z,relro -Wl,-z,now -Wl,--as-needed     -L/usr/armv7l-linux-gnueabihf/usr/lib' '-Dldflag
s=-specs=/void-packages/common/environment/configure/gccspecs/hardened-ld -Wl,-z,relro -Wl,-z,now -Wl,--as-needed     -L/usr/armv7l-linux-gnueabihf/usr/lib' '-Dperl_static_inline=static __inline__' -Dd_static_inline

...

```
Set user cccdlflags='-fPIC'                                                     Set user optimize=' -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2  -O2 -pipe -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard   -I/usr/armv7l-linux-gnueabihf/usr/include '
Set user ccflags=' -specs=/void-packages/common/environment/configure/gccspecs/h
ardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2  -O2 -pipe -march=armv7-a -mfpu=vfpv3 -mfloat-abi=hard   -I/usr/armv7l-linux-gnueabihf/usr/include '
Set user lddlflags='-shared -specs=/void-packages/common/environment/configure/gccspecs/hardened-ld -Wl,-z,relro -Wl,-z,now -Wl,--as-needed     -L/usr/armv7l-linux-gnueabihf/usr/lib'
Set user ldflags='-specs=/void-packages/common/environment/configure/gccspecs/ha
rdened-ld -Wl,-z,relro -Wl,-z,now -Wl,--as-needed     -L/usr/armv7l-linux-gnueabihf/usr/lib'                
...
Checking whether to include <stdio.h>                                           
| #include <stdio.h>                                                            
> gcc -D_POSIX_C_SOURCE -D_XOPEN_SOURCE -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64-W -c -o try.o try.c
```

**Comment by chneukirchen** [comment]
Trying to strip it down, somehow it only happens in my building environment.

**Comment by chneukirchen** [comment]
Ok, the -W issue is fixed in 131308cf6 already.

**Comment by chneukirchen** [comment]
And the prefixing too.  Sorry.  Perhaps a new release is good. ;)
